File: code/coroutines-tasks.cs
var client = new HttpClient();

var repos = new [] { "jetbrains/kotlin", "dotnet/csharplang" };

var asyncRequests = repos.Select(async repo =>
{
    var response = await client.GetAsync($"https://api.github.com/repos/{repo}");
    var json = await response.Content.ReadAsStringAsync();
    dynamic content = JsonConvert.DeserializeObject(json);
    return new { repo, stars = content.stargazers_count, forks = content.forks };
});

var results = await Task.WhenAll(asyncRequests);

foreach(var data in results)
    Console.WriteLine($"{data.repo} : {data.stars} - {data.forks}");

File: code/coroutines-tasks.kt
data class Stats(val full_name: String, val stargazers_count: Int = -1, val forks: Int = -1)

val mapper = jacksonObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)

val repos = listOf("jetbrains/kotlin", "dotnet/csharplang")

val asyncRequests = repos.map { repo ->
    GlobalScope.async {
        val body = Fuel.get("https://api.github.com/repos/${repo}")
            .responseString()
            .third.component1() // Fuel Result & Body
        body?.let { mapper.readValue<Stats>(it) } ?: Stats(repo)
    }
}

runBlocking {
    val results = asyncRequests.map { it.await() }
    results.forEach{ println("${it.full_name} : ${it.stargazers_count} - ${it.forks}") }
}